notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb

```diff
@@ -0,0 +1,472 @@
+    "# To allow display tweaks for wider response tables\n",
+    "from IPython.core.display import display, HTML\n",
+    "\n",
+    "# For the second example: kernel density estimates\n",
+    "from scipy.stats import gaussian_kde"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Display Tweaks\n",
+    "<div style=\"text-align: right\">[(top)](#title_ID)</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set page width to fill browser for longer output lines\n",
+    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))\n",
+    "\n",
+    "# set width for pprint\n",
+    "astropy.conf.max_width = 150"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "***\n",
+    "## Connecting, Finding, and Displaying Table Information \n",
+    "<div style=\"text-align: right\">[(top)](#MAST-TAP-Hubble-Source-Catalog-Demo)</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Connecting to a TAP Service\n",
+    "<div style=\"text-align: right\">[(top)](#title_ID)</div>\n",
+    "\n",
+    "The TapPlus library is able to connect to any TAP service, given the \"base\" URL as noted in metadata registry resources describing the service. This is the URL for the newest version of the Hubble Source Catalog TAP service."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "HSC_service = TapPlus(url=\"http://vao.stsci.edu/HSCTAP/tapservice.aspx\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Querying for Table Schema Information\n",
+    "<div style=\"text-align: right\">[(top)](#title_ID)</div>\n",
+    "    \n",
+    "TAP services are self-describing, which means the service itself can be asked for its schema and documentation about it. Since the hubble Source Catalog does not follow a data model described by a standard, this is the best way to see what tables and columns we have available to then query based on geometry or other filters. \n",
+    "    \n",
+    "The main view for HSC, SumMagAper2CatView, is extremely wide, containing columns for all potential filters, each of which may have null data. So in showing our query results, we will cut off the display with \"...\" marks. You can change the 'if' line to show the rest of these columns."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "HSC_tables = HSC_service.load_tables()\n",
+    "print('\\n')\n",
+    "for table in HSC_tables:\n",
+    "    if( table.get_name() == 'dbo.SumMagAper2CatView'):\n",
+    "        print(table)\n",
+    "        print('\\n')\n",
+    "        columns = table.get_columns()\n",
+    "        for i, column in enumerate(columns):\n",
```

Comment:
This is a suggestion you should feel free to ignore if you disagree... but the following might be clearer/more pythonic instead of using the for loop with index checking:
```
        for column in columns[:30]:
            print(column.get_name())
        print('...')
        for column in columns[-10:]:
            print(column.get_name())
```

(making that change would then require a bit of change to the text above)

Reply:
I'm going to ignore this one for now to avoid making text changes where we don't have to but I can be convinced otherwise.